addon/components/layers-dialogs/settings/wfs.js

```diff
@@ -4,46 +4,128 @@
+import WmsSettingsComponent from './wms';
+
+// Regular expression used to derive whether settings' url is correct.
+let urlRegex = '(https?|ftp)://(-\.)?([^\s/?\.#-]+\.?)+(/[^\s]*)?';
 
 /**
   Settings-part of WFS layer modal dialog.
 
   @class WfsLayerSettingsComponent
-  @extends <a href="http://emberjs.com/api/classes/Ember.Component.html">Ember.Component</a>
+  @extends WmsSettingsComponent
 */
-export default Ember.Component.extend({
+export default WmsSettingsComponent.extend({
   /**
     Array containing available formats.
 
-    @property _availableFormats
+    @property _availableInfoFormats
     @type String[]
     @private
   */
-  _availableFormats: null,
+  _availableInfoFormats: null,
 
   /**
     Reference to component's template.
   */
   layout,
 
   /**
-    Overridden ['tagName'](http://emberjs.com/api/classes/Ember.Component.html#property_tagName)
-    is empty to disable component's wrapping <div>.
+    Get capabilities button error message.
 
-    @property tagName
+    @property getCapabilitiesErrorMessage
     @type String
-    @default ''
+    @readonly
+  */
+  getCapabilitiesErrorMessage: Ember.computed(
+    'getCapabilitiesPromiseError',
+    'i18n',
+    'settings.url',
+    'settings.typeNS',
+    'settings.typeName',
+    'settings.version',
+    function () {
+      let getCapabilitiesPromiseError = this.get('getCapabilitiesPromiseError');
+
+      if (!Ember.isBlank(getCapabilitiesPromiseError)) {
+        this.set('getCapabilitiesPromiseError', null);
+        return getCapabilitiesPromiseError;
+      }
+
+      let message;
+      let fields = Ember.A();
+
+      let i18n = this.get('i18n');
+
+      let url = this.get('settings.url');
+      let typeNS = this.get('settings.typeNS');
+      let typeName = this.get('settings.typeName');
+      let version = this.get('settings.version');
+
+      if (Ember.isBlank(url) || Ember.isBlank(url.toString().match(new RegExp(urlRegex)))) {
+        fields.pushObject(i18n.t('components.layers-dialogs.settings.wfs.url-textbox.caption'));
+      }
+
+      if (Ember.isBlank(typeName)) {
+        fields.pushObject(i18n.t('components.layers-dialogs.settings.wfs.type-name-textbox.caption'));
+      }
+
+      if (Ember.isBlank(typeNS)) {
+        fields.pushObject(i18n.t('components.layers-dialogs.settings.wfs.type-ns-textbox.caption'));
+      }
+
+      if (Ember.isBlank(version)) {
+        fields.pushObject(i18n.t('components.layers-dialogs.settings.wfs.version-textbox.caption'));
+      }
+
+      if (!Ember.isBlank(fields)) {
+        message = i18n.t('components.layers-dialogs.edit.get-capabilities-button.error-caption') + fields.join(', ');
+      }
+
+      return message;
+    }
+  ),
+
+  /**
+    OBSOLETE
+    Observes changes in settings.url property and checks whether it is correct url.
+    If url syntax is correct, requests available info formats from service.
+
+    @method _urlDidChange
+    @private
   */
-  tagName: '',
```

Comment:
do you need div tag in this component?

Reply:
it is inherited from WmsSettingsComponent where it is overriden and empty

Reply:
ok